Scripts/artwork.gd

```diff
@@ -0,0 +1,18 @@
+@tool
+extends Area2D
+
+@export var _texture : Texture2D :
+	set(value):
+		_texture = value
+		
+		if is_node_ready() == false:
+			# make sure node is ready so @onready vars are set
+			await ready
+		
+		# update sprite and collider
+		_sprite.texture = _texture
+		if value != null:
+			_collider.shape.size = _texture.get_size()
+
```

Comment:
I think we'll need to add a boolean value to check if the player has collected this artwork or not. im not 100% sure how godot works with this but im thinking something like a manager class that assigns it on scene load. and those values also needed to be saved somewhere

Reply:
yeah I'm thinking a global object (Autoload) that keeps track of things like this which are saved to file when player closes the game so they can keep the progress next time they play, I don't think it's this class's responsibility though

Reply:
do you mean auto load them into scene? my brain was thinking something simpler like just have all these objects in the scene already then we just tick them on/off to render the texture if user has collected it.

Reply:
that's also an option, what I'm proposing is that we'd have a class that keeps track of progress (unlocked levels, art gathered etc..) so we can save and retrieve data from disk, and based on this object the Gallery scene would at the start read the data (art pieces gathered in this case) and show/hide them accordingly, we can also have the artworks check the data and toggle themselves, I just think having this behavior in one place for all of them would be easier to manage.